### DISCUSSION TITLE
Writing test cases for STM32F407VGTx

### DISCUSSION THREAD

**Discussion by Admsher** [stm32-rs/stm32f4xx-hal#822] [Q&A] [answered]
I am trying to write a test case framework for my project that runs on STM32F407VGT6 board. I tried defmt-test for this but i keep getting looed in an error for missing rust-lld, is there by any chance a working project for the same that i can follow here is my project repo if needed : https://github.com/Admsher/EMDU_pc any help is appreciated thanks in advance

**Comment by burrbull** [comment, marked as answer]
There is also https://crates.io/crates/embedded-test

> missing rust-lld

This should be installed by `rustup target add thumbv7em-none-eabihf`

Try also `rustup component add llvm-tools`. Although it should not be related.

**Reply by Admsher** [reply]
I already have the target binaries installed, but still the same will try the component binary and update here

**Reply by Admsher** [reply]
Update : 
The embedded-test crate works rather than defmt-test, just be careful to change the hal in the example to the hal of stm32
